h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -345,13 +346,27 @@ public void computeCrossValidation() {
       // scores; compute gains/lifts
       cv_mainModelScores(N, mbs, cvModelBuilders);
 
-      // Step 8: Clean up potentially created temp frames
-      for (ModelBuilder mb : cvModelBuilders)
-        mb.cleanUp();
-
       _job.setReadyForView(true);
       DKV.put(_job);
+    } catch (Exception e) {
+      if (cvModelBuilders != null) {
+        Futures fs = new Futures();
```

Comment:
you should probably wait for completion of the tasks you put in `fs` at the end for the block (fs.blockForPending()). Otherwise, race conditions might occur.

Reply:
oh right, forgot it, thx!